connector/datadogconnector/config.go

```diff
@@ -0,0 +1,83 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package datadogconnector // import "github.com/open-telemetry/opentelemetry-collector-contrib/connector/datadogconnector"
+
+import (
+	"fmt"
+	"regexp"
+
+	"go.opentelemetry.io/collector/component"
+)
+
+var _ component.Config = (*Config)(nil)
+
+type Config struct {
```

Comment:
Is there a way to share config between connector & exporter to avoid duplicate code ? like use internal/datadog to create trace agent for both agent & connector

Reply:
That will be large refactor, I'd prefer not to do it until we decide the path forward for datadogconnector.

Reply:
We can do this via embedding which would avoid a breaking change, so I am fine waiting on this